sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/TestTextLine.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.common;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import org.apache.beam.sdk.transforms.DoFn;
+
+/**
+ * Used to pass values around within test pipelines for text file based IO.
+ */
+public abstract class TestTextLine {
```

Comment:
<!--new_thread; commit:c6c7070ad92424707d3720d3a4dc2c0fb6961440; resolved:0-->
I think this class can be stripped down to a small anonymous class inside TextIOIT. Is there a need to make this a separate class ?

Reply:
My motivation is as follows: this is a step towards reusing this class' code in other future tests, eg. those for Compressed Text, AvroIO etc. It looks like we are going to need it here in the future. Also this class is in fact is very alike to TestRow class which seems to be created for database IO tests accordingly. Would that be ok if i leave it as is?

Reply:
Ok, on the second thought I see that when we rename module `text` to `file-based-io-tests` we can move the class there to be reused by all file based io tests.